lib/azure/armrest/resource_provider_service.rb

```diff
@@ -53,7 +54,8 @@ def list
       def get(namespace)
         url = build_url(namespace)
         response = rest_get(url)
-        JSON.parse(response.body)
+        #JSON.parse(response.body)
+        Azure::Armrest::ResourceProvider.new(response.body)
```

Comment:
If the response JSON string from the API contains an array at root level, how do you handle? You don't want to use `JSON.parse` to get an array of hashes, and use `.map` to convert each back to a JSON string, do you?